### ISSUE TITLE
Accessibility violations prompted by the bx-tab from the tabs Web Component

### ISSUE THREAD

**Issue by jc-corrales** [carbon-design-system/carbon-for-ibm-dotcom#10921]
### Description

When scanning the bx-tabs with the IBM Equal Access Accessibility checker, it will throw this error for the a tag within the ShadowRoot:

`Interactive component with ARIA role 'tab' does not have a programmatically associated name
Interactive component must have a programmatically associated name`

I am seeing this also in my project. And because it is prompted by an A tag from within the shadowRoot, the only potential workaround I can see is by direct override via JS, which I don't feel would be a best practice.

### Component(s) impacted

Tabs

### Browser

Chrome, Microsoft Edge

### Carbon for IBM.com version

v1.31.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Legal Entity Repository, can be found in storybook itself

### Package

@carbon/web-components

### CodeSandbox example

Don't have

### Steps to reproduce the issue (if applicable)

1. Open storybook "https://web-components-v1.carbondesignsystem.com/?path=/story/introduction-welcome--page"
2. Open the tabs component
3. Run the IBM Equal Access Accessibility Checker
4. Identify errors that say "The tabbable element's role 'none' is not a widget role"

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by pjudge** [comment]
Hi! Can I be assigned to this issue? I'd love to help out. Thanks
